pywinauto/controls/hwndwrapper.py

```diff
@@ -1173,68 +1173,17 @@ def set_focus(self):
-        if not cur_foreground:
-            mouse.move(coords=(10000, 20000))
-            win32gui.SetForegroundWindow(self.handle)
-            win32functions.WaitGuiThreadIdle(self)
-            time.sleep(Timings.after_setfocus_wait)
-
-        # "steel the focus" if there is another active window
-        # otherwise it is already into the foreground and no action required
-        elif self.handle != cur_foreground:
-            mouse.move(coords=(10000, 20000))
-
-            # get the thread of the window that is in the foreground
-            cur_fore_thread = win32process.GetWindowThreadProcessId(
-                cur_foreground)[0]
-
-            # get the thread of the window that we want to be in the foreground
-            control_thread = win32process.GetWindowThreadProcessId(
-                self.handle)[0]
-
-            # if a different thread owns the active window
-            if cur_fore_thread != control_thread:
-
-                # Attach the two threads and set the foreground window
-                win32process.AttachThreadInput(control_thread,
-                                               cur_fore_thread,
-                                               1)
-
-                win32gui.SetForegroundWindow(self.handle)
-
-                # ensure foreground window has changed to the target
-                # or is 0(no foreground window) before the threads detaching
-                timings.wait_until(
-                    Timings.setfocus_timeout,
-                    Timings.setfocus_retry,
-                    lambda: win32gui.GetForegroundWindow()
-                    in [self.top_level_parent().handle, 0])
-
-                # get the threads again to check they are still valid.
-                cur_fore_thread = win32process.GetWindowThreadProcessId(
-                    cur_foreground)[0]
-                control_thread = win32process.GetWindowThreadProcessId(
-                    self.handle)[0]
-
-                if cur_fore_thread and control_thread:  # both are valid
-                    # Detach the threads
-                    win32process.AttachThreadInput(control_thread,
-                                                   cur_fore_thread,
-                                                   0)
-            else:
-                # same threads - just set the foreground window
-                win32gui.SetForegroundWindow(self.handle)
+        # change active window
+        win32gui.ShowWindow(self.handle, win32con.SW_RESTORE)
+        win32gui.SetForegroundWindow(self.handle)
 
-            # make sure that we are idle before returning
-            win32functions.WaitGuiThreadIdle(self)
+        # make sure that we are idle before returning
+        win32functions.WaitGuiThreadIdle(self)
 
-            # only sleep if we had to change something!
-            time.sleep(Timings.after_setfocus_wait)
+        # only sleep if we had to change something!
+        time.sleep(Timings.after_setfocus_wait)
```

Comment:
@mtkennerly, thanks for the PR. Even if we decide to stop stealing focus with ```AttachThreadInput``` it's still required to verify if the window already has a focus. I think that there is no point in running ``` ShowWindow/SetForeground/time.sleep``` sequence bluntly. I would also still set mouse move coordinates different from the left bottom corner.
```python
cur_foreground = win32gui.GetForegroundWindow()
if self.handle != cur_foreground:
    mouse.move(coords=(-10000, 500))  # move the mouse out of screen to the right
    win32gui.ShowWindow(self.handle, win32con.SW_RESTORE)
    win32gui.SetForegroundWindow(self.handle)
    win32functions.WaitGuiThreadIdle(self)
    time.sleep(Timings.after_setfocus_wait)
```